autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/ExistingPasswordMatchDetector.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.impl.importing
+
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.impl.store.InternalAutofillStore
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.withContext
+
+interface ExistingPasswordMatchDetector {
+    suspend fun alreadyExists(newCredentials: LoginCredentials): Boolean
+}
+
+@ContributesBinding(AppScope::class)
+class DefaultExistingPasswordMatchDetector @Inject constructor(
+    private val autofillStore: InternalAutofillStore,
+    private val dispatchers: DispatcherProvider,
+) : ExistingPasswordMatchDetector {
+
+    override suspend fun alreadyExists(newCredentials: LoginCredentials): Boolean {
+        return withContext(dispatchers.io()) {
+            val credentials = autofillStore.getAllCredentials().firstOrNull() ?: return@withContext false
```

Comment:
This method is being called once for every entry on the imported file, and we are accessing db and retrieving allCredentials every time. That can be several calls to db for an import, can we improve this part?